crates/iota/src/validator_commands.rs

```diff
@@ -1014,29 +1027,104 @@ pub async fn get_validator_summary(
             (summary.iota_address, (summary, committee_member))
         })
         .collect::<BTreeMap<_, _>>();
-    let validator_info =
-        if let Some((_, committee_member)) = active_validators.get(&validator_address) {
-            if *committee_member {
-                status = Some(ValidatorStatus::CommitteeMember);
-            } else {
-                status = Some(ValidatorStatus::Active);
-            }
-            Some(active_validators.remove(&validator_address).unwrap().0)
-        } else {
-            // Check pending validators
-            get_pending_candidate_summary(validator_address, client, pending_active_validators_id)
-                .await?
-                .map(|v| v.into_iota_validator_summary())
-                .tap_some(|_s| status = Some(ValidatorStatus::Pending))
 
-            // TODO also check candidate and inactive validators
+    // Check active and committee members
+    if let Some((_, committee_member)) = active_validators.get(&validator_address) {
```

Comment:
Since `active_validators` is not used anymore after this, you could match on `remove` instead of `get`, gets rid of the unwrap

Reply:
https://github.com/iotaledger/iota/pull/6137/commits/7a612ab8aa46612f7b5c6204d20354d33039efd1